api/wasm.go

```diff
@@ -393,6 +393,92 @@ func DecodeF64(input uint64) float64 {
 	return math.Float64frombits(input)
 }
 
+// EncodeI8x16 encodes the input as a ValueTypeV128.
+func EncodeI8x16(i1, i2, i3, i4, i5, i6, i7, i8, i9, i10, i11, i12, i13, i14, i15, i16 int8) (low uint64, hi uint64) {
```

Comment:
I suspect that when dealing with bytes, it will be more common to be working with `[]byte`. Would we provide another function for it?

Reply:
Actually, I think for all of them (except maybe 64-bit), slice will be common, e.g. 

```go
var arr []byte
for i := 0; i < len(arr); i += 16 {
  low, hi := EncodeI8x16(arr[i:i+16])
  func.Call(low, hi)
}
```

```go
var arr []int16
for i := 0; i < len(arr); i += 8{
  low, hi := EncodeI16x18(arr[i:i+8])
  func.Call(low, hi)
}
```

Not sure if we should have both slice and non-slice versions but I think slice are important.

Reply:
Ah I guess if `arr[i:i+16]...` works it's all good, oops

Reply:
yeah will change the signature!